### ISSUE TITLE
More than 8 bits per sample for vdif

### ISSUE THREAD

**Issue by jswoboda** [mhvk/baseband#527]
I was trying to write write 16 bit complex samples to vdif but there baseband fails when I try to write something greater than 8 bits:

`ValueError: VDIFPayload cannot encode data with 16 bits`

From what I understand given the [vdiff spec](https://vlbi.org/wp-content/uploads/2019/03/VDIF_specification_Release_1.1.1.pdf) is that there is a maximum of 32 bits per sample. The highest number of bits per sample is 8 that I've been able to get to work.

Given the rise in RFI it would be a good idea have the capability to deal with higher bit depths.

**Comment by mhvk** [comment]
Indeed, VDIF supports it but beyond the data formats supported by [mark5access](https://github.com/demorest/mark5access), I tend to add formats only as I encounter data sets that need them, so 16 bit is not yet there. I can add it, though, if it is useful to you -- even better if you have a small actual data sample to base the tests on. Or, if you feel adventurous, make a PR adding an encoder and decoder to https://github.com/mhvk/baseband/blob/main/baseband/base/encoding.py
and then use them in https://github.com/mhvk/baseband/blob/main/baseband/vdif/payload.py
